### PR TITLE
apply channels and streaming to the url cache

### PR DESCRIPTION
**Pull request by ryanfkeepers**
updates the url cache item enumeration with stream processing
using channels.  Plus other changes to accomodate the
new pattern such as mock and interface updates. 

This is the second of a multipart update that has been separated
for ease of review.  CI is not expected to pass until the final
PR.

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :broom: Tech Debt/Cleanup

#### Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/m365/service/onedrive/mock/handlers.go [hunk 1]
```diff
@@ -9,23 +9,25 @@ import (
...
 
 // ---------------------------------------------------------------------------
 // Backup Handler
 // ---------------------------------------------------------------------------
 
-type BackupHandler struct {
+type BackupHandler[T any] struct {
 	ItemInfo details.ItemInfo
+	// FIXME: this is a hacky solution.  Better to use an interface
+	// and plug in the selector scope there.
```
Comment (ashmrtn) [inline src/internal/m365/service/onedrive/mock/handlers.go:27]: how about we make an interface that selectors implement?

**Review by ashmrtn — APPROVED** [review]

File: src/internal/m365/collection/drive/url_cache.go [hunk 2]
```diff
@@ -157,19 +157,29 @@ func (uc *urlCache) refreshCache(
...
-		api.URLCacheDriveItemProps())
-	if err != nil {
-		uc.idToProps = make(map[string]itemProps)
-		return clues.Stack(err)
+		api.CallConfig{
+			Select: api.URLCacheDriveItemProps(),
+		})
+
+	for page, reset, done := pager.NextPage(); !done; {
+		uc.deltaQueryCount++
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/url_cache.go:169]: Ah, wasn't sure what we wanted there.  The var name isn't too clear.  Thanks.

File: src/internal/m365/service/onedrive/mock/handlers.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/m365/service/onedrive/mock/handlers.go:27]: Yup, that's the plan.

File: src/internal/m365/service/onedrive/mock/handlers.go [hunk 3]
```diff
@@ -9,23 +9,25 @@ import (
...
 	"github.com/alcionai/corso/src/pkg/services/m365/api"
-	apiMock "github.com/alcionai/corso/src/pkg/services/m365/api/mock"
 )
 
 // ---------------------------------------------------------------------------
 // Backup Handler
 // ---------------------------------------------------------------------------
 
-type BackupHandler struct {
+type BackupHandler[T any] struct {
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/m365/service/onedrive/mock/handlers.go:24]: It doesn't, and I'm already removing that in the next PR.

File: src/internal/m365/collection/drive/url_cache_test.go [hunk 4]
```diff
@@ -196,39 +204,140 @@ func (suite *URLCacheUnitSuite) TestGetItemProperties() {
...
+			},
+		},
+		{
+			name: "multiple pages with resets",
+			pages: []mock.NextPage[models.DriveItemable]{
+				{
+					Items: []models.DriveItemable{
+						fileItem("1", "file1", "root", "root", "https://dummy1.com", false),
+						fileItem("2", "file2", "root", "root", "https://dummy2.com", false),
+						fileItem("3", "file3", "root", "root", "https://dummy3.com", false),
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/m365/collection/drive/url_cache_test.go:322]: Good idea!

**Pull request merged**